pkg/converter/target/externalentity.go

```diff
@@ -66,20 +70,65 @@ func NewExternalEntityFrom(
 
 // PatchExternalEntityFrom generate a patch for existing ExternalEntity from source.
 func PatchExternalEntityFrom(
-	source ExternalEntitySource, patch *antreatypes.ExternalEntity, cl client.Client) *antreatypes.ExternalEntity {
-	populateExternalEntityFrom(source, patch, cl)
-	return patch
+	source ExternalEntitySource, patch *antreatypes.ExternalEntity, cl client.Client) (*antreatypes.ExternalEntity, bool) {
+	changed := populateExternalEntityFrom(source, patch, cl)
+	return patch, changed
 }
 
 func populateExternalEntityFrom(source ExternalEntitySource, externalEntity *antreatypes.ExternalEntity,
-	cl client.Client) {
-	externalEntity.SetLabels(genTargetEntityLabels(source, cl))
+	cl client.Client) bool {
+	changed := false
+	if !reflect.DeepEqual(externalEntity.GetLabels(), genTargetEntityLabels(source, cl)) {
+		externalEntity.SetLabels(genTargetEntityLabels(source, cl))
+		changed = true
+	}
 	ipAddrs, _ := source.GetEndPointAddresses()
 	endpoints := make([]antreatypes.Endpoint, 0, len(ipAddrs))
 	for _, ip := range ipAddrs {
 		endpoints = append(endpoints, antreatypes.Endpoint{IP: ip})
 	}
-	externalEntity.Spec.Endpoints = endpoints
-	externalEntity.Spec.Ports = source.GetEndPointPort(cl)
-	externalEntity.Spec.ExternalNode = source.GetExternalNodeName(cl)
+	if externalEntity.Spec.ExternalNode != source.GetExternalNodeName(cl) {
+		externalEntity.Spec.ExternalNode = source.GetExternalNodeName(cl)
+		changed = true
+	}
+	sourcePorts := source.GetEndPointPort(cl)
+	sort.Slice(externalEntity.Spec.Ports, func(i, j int) bool {
+		if externalEntity.Spec.Ports[i].Name != externalEntity.Spec.Ports[j].Name {
+			return strings.Compare(externalEntity.Spec.Ports[i].Name, externalEntity.Spec.Ports[j].Name) < 0
+		}
+		if externalEntity.Spec.Ports[i].Port != externalEntity.Spec.Ports[j].Port {
+			return externalEntity.Spec.Ports[i].Port < externalEntity.Spec.Ports[j].Port
+		}
+		return strings.Compare(string(externalEntity.Spec.Ports[i].Protocol), string(externalEntity.Spec.Ports[j].Protocol)) < 0
+	})
+	sort.Slice(sourcePorts, func(i, j int) bool {
+		if sourcePorts[i].Name != sourcePorts[j].Name {
+			return strings.Compare(sourcePorts[i].Name, sourcePorts[j].Name) < 0
+		}
+		if sourcePorts[i].Port != sourcePorts[j].Port {
+			return sourcePorts[i].Port < sourcePorts[j].Port
+		}
+		return strings.Compare(string(sourcePorts[i].Protocol), string(sourcePorts[j].Protocol)) < 0
+	})
+	if !reflect.DeepEqual(externalEntity.Spec.Ports, sourcePorts) {
+		externalEntity.Spec.Ports = sourcePorts
+		changed = true
+	}
+	sort.Slice(externalEntity.Spec.Endpoints, func(i, j int) bool {
+		if externalEntity.Spec.Endpoints[i].Name != externalEntity.Spec.Endpoints[j].Name {
```

Comment:
Same as above. Please see if you can define a comparator function to sort Endpoints?

Reply:
Added